src/System.Private.CoreLib/src/System.Private.CoreLib.csproj

```diff
@@ -277,11 +279,8 @@
     <Compile Include="System\Threading\ManagedThreadId.cs" />
     <Compile Include="System\Threading\Lock.cs" />
     <Compile Include="System\Threading\Condition.cs" />
-    <Compile Include="System\Threading\FirstLevelSpinWaiter.cs" />
     <Compile Include="System\Threading\Interlocked.cs" />
     <Compile Include="System\Threading\LockHolder.cs" />
-    <Compile Include="System\Threading\LowLevelLock.cs" />
-    <Compile Include="System\Threading\LowLevelMonitor.cs" />
```

Comment:
IIRC, people working on this found it convenient to be able to debug the WaitSystem on Windows. That's whe LowLevelMonitor was implemented on Unix.

cc @kouvel @jkoritzinsky

Reply:
Interesting. I thought the point was to support debugging ThreadPool.Portable on Windows (the only usage outside of WaitSubsystem). That is still supported after this change. The whole WaitSubsystem was already guarded in the same Unix-only conditions.

I was more worried about introducing some performance bottleneck by replacing usages of LowLevelLock with Lock, but I didn't get to run the benchmarks yet to verify the impact.

If it turns out to be a problem I can revert the last few commits and take a different approach with bringing some implementation of LowLevelLock (and LowLevelSemaphore) to Mono.

Reply:
It would be useful to maintain being able to use the portable thread pool on Windows as well under at least a build-time flag for debugging and perf comparisons. There shouldn't be too many dependencies other than the few that were already identified here.

Reply:
> It would be useful to maintain being able to use the portable thread pool on Windows as well under at least a build-time flag for debugging and perf comparisons. 

Definitely. I use that myself.